pkg/key/key.go

```diff
@@ -100,12 +101,19 @@ func CloudFrontDistributionComment(clusterID string) string {
 	return fmt.Sprintf("Created by irsa-operator for cluster %s", clusterID)
 }
 
-func CloudFrontAlias(clusterID string, installation string, region string) string {
-	return fmt.Sprintf("irsa.%s", BaseDomain(clusterID, installation, region))
+func CloudFrontAlias(baseDomain string) string {
+	return fmt.Sprintf("irsa.%s", baseDomain)
 }
 
-func BaseDomain(clusterID string, installation string, region string) string {
-	return fmt.Sprintf("%s.k8s.%s.%s.aws.gigantic.io", clusterID, installation, region)
+func BaseDomain(cluster capi.Cluster) (string, error) {
+	apiEndpoint := cluster.Spec.ControlPlaneEndpoint.Host
```

Comment:
And I'm not even sure when controller is filling out the ControlPlaneEndpoint. This might be tricky.

Cluster.Spec.ControlPlaneEndpoint.Host is filled out by aws-operator. But aws-operator is also waiting for irsa configmap.

Reply:
controlplaneendoint.host is set almost immediately after cluster creation

Reply:
nice, so we only have to think about CAPA, what we use as a basedomain

Reply:
yes that's going to be a problem when I'll work on the CAPA PR (coming up next)